### DISCUSSION TITLE
Finding delay for different parts of the path

### DISCUSSION THREAD

**Discussion by weirdooo-18** [NVlabs/sionna#364] [Q&A] [answered]
Hello there!!
Is there any way to find the delay or any other parameters for the first part of the path, say, until the ray experiences the first reflection?
Say, if there are two reflections, is there a way to get the propagation delay experienced by it before the first reflection, after the first reflection and after the second reflection?

Thank you.

**Comment by jhoydis** [comment, marked as answer]
Hi,

As already mentioned in my response to #360, the `Paths` object has a property called vertices which contains the sequence of all intersection points of a ray with the scene. You can simply compute the distances between any two intersection points and divide them by the speed of light. This would give you the delays for all path segments.

Please have a look at the source code of this function `_compute_directions_distances_delays_angles` in the file [solver_paths.py](https://github.com/NVlabs/sionna/blob/main/sionna/rt/solver_paths.py).

**Reply by weirdooo-18** [reply]
Thank you so much for the clarification.
